test/cli/search_test.cpp

```diff
@@ -0,0 +1,454 @@
+#include <string>
+
+#include "cli_test.hpp"
+
+struct search_test : public cli_test
+{
+    void run_search_test(std::string const & index_command,
+                         std::string const & db_file,
+                         std::string const & index_file,
+                         std::string const & index_type,
+                         std::string const & reduction,
+                         std::string const & search_command,
+                         std::string const & query_file,
+                         std::string const & output_file,
+                         std::string const & output_type,
+                         std::string const & control_file)
+    {
+        cli_test_result result_index = execute_app("lambda3", index_command,
+                                                   "-d", data(db_file),
+                                                   "-i", index_file,
+                                                   "--db-index-type", index_type,
+                                                   "-r", reduction);
+
+        cli_test_result result_search = execute_app("lambda3", search_command,
+                                                    "-i", index_file,
+                                                    "-q", data(query_file),
+                                                    "-t", "2",
+                                                    "--version-to-outputfile", "0",
+                                                    "-o", output_file);
+
+        ASSERT_EQ(result_search.exit_code, 0);
+
+        if (output_type == "bam")
+        {
+            std::system(("md5sum " + output_file + " > md5sum_test.txt").c_str());
+            std::system(("md5sum " + (std::string) data(control_file) + " > md5sum_control.txt").c_str());
+
+            std::ifstream test_output ("md5sum_test.txt");
+            std::ifstream control_output ("md5sum_control.txt");
+
+            std::string test_line;
+            std::string control_line;
+
+            getline(test_output, test_line);
+            getline(control_output, control_line);
+
+            std::string test_sum = test_line.substr(0, test_line.find(' '));
+            std::string control_sum = control_line.substr(0, control_line.find(' '));
+
+            ASSERT_EQ(test_sum, control_sum);
+        }
+        else
+        {
+            std::ifstream test_output;
+
+            if (output_type == "m9_gz")
+            {
+                system(("gunzip " + output_file).c_str());
+                test_output.open(output_file.substr(0, output_file.length() - 3));
+            }
+            else if (output_type == "sam_bz2")
+            {
+                system(("bzip2 -d " + output_file).c_str());
+                test_output.open(output_file.substr(0, output_file.length() - 4));
+            }
+            else
+            {
+                test_output.open(output_file);
+            }
+
+            std::ifstream control_output (data(control_file));
```

Comment:
```suggestion
            std::ifstream control_output{control_file};
```
?

Reply:
We need the data() so that it is clear that this is not located in the test output directory but in the data directory. Otherwise the test will fail as it can't find the control file in the test output directory.